File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-ird-5-4/SwapAdditionalTerms.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class SwapAdditionalTerms : ISerialized
    {
        public SwapAdditionalTerms(XmlNode xmlNode)
        {
            XmlNode bondReferenceNode = xmlNode.SelectSingleNode("bondReference");
            
            if (bondReferenceNode != null)
            {
                if (bondReferenceNode.Attributes["href"] != null || bondReferenceNode.Attributes["id"] != null) 
                {
                    if (bondReferenceNode.Attributes["id"] != null) 
                    {
                        bondReferenceIDRef_ = bondReferenceNode.Attributes["id"].Value;
                        BondReference ob = new BondReference(bondReferenceNode);
                        IDManager.SetID(bondReferenceIDRef_, ob);
                    }
                    else if (bondReferenceNode.Attributes["href"] != null)
                    {
                        bondReferenceIDRef_ = bondReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        bondReference_ = new BondReference(bondReferenceNode);
                    }
                }
                else
                {
                    bondReference_ = new BondReference(bondReferenceNode);
                }
            }
            
        
        }
        
    
        #region BondReference_
        private BondReference bondReference_;
        public BondReference BondReference_
        {
            get
            {
                if (this.bondReference_ != null)
                {
                    return this.bondReference_; 
                }
                else if (this.bondReferenceIDRef_ != null)
                {
                    bondReference_ = IDManager.getID(bondReferenceIDRef_) as BondReference;
                    return this.bondReference_; 
                }
                else
                {
                      return this.bondReference_; 
                }
            }
            set
            {
                if (this.bondReference_ != value)
                {
                    this.bondReference_ = value;
                }
            }
        }
        #endregion
        
        public string bondReferenceIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-ird-5-4/SwapAdditionalTerms.cpp
// SwapAdditionalTerms.cpp 
#include "SwapAdditionalTerms.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

SwapAdditionalTerms::SwapAdditionalTerms(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //bondReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* bondReferenceNode = xmlNode->FirstChildElement("bondReference");

   if(bondReferenceNode){bondReferenceIsNull_ = false;}
   else{bondReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- bondReferenceNode , address : " << bondReferenceNode << std::endl;
   #endif
   if(bondReferenceNode)
   {
      if(bondReferenceNode->Attribute("href") || bondReferenceNode->Attribute("id"))
      {
          if(bondReferenceNode->Attribute("id"))
          {
             bondReferenceIDRef_ = bondReferenceNode->Attribute("id");
             bondReference_ = boost::shared_ptr<BondReference>(new BondReference(bondReferenceNode));
             bondReference_->setID(bondReferenceIDRef_);
             IDManager::instance().setID(bondReferenceIDRef_,bondReference_);
          }
          else if(bondReferenceNode->Attribute("href")) { bondReferenceIDRef_ = bondReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { bondReference_ = boost::shared_ptr<BondReference>(new BondReference(bondReferenceNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<BondReference> SwapAdditionalTerms::getBondReference()
{
   if(!this->bondReferenceIsNull_){
        if(bondReferenceIDRef_ != ""){
             return boost::shared_static_cast<BondReference>(IDManager::instance().getID(bondReferenceIDRef_));
        }else{
             return this->bondReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<BondReference>();
   }
}
}

